ui/components/multichain/create-named-snap-account/create-named-snap-account.tsx

```diff
@@ -33,44 +40,40 @@ export const CreateNamedSnapAccount: React.FC<CreateNamedSnapAccountProps> = ({
 }) => {
   const t = useI18nContext();
   const history = useHistory();
-  const dispatch = useDispatch();
 
   const mostRecentOverviewPage = useSelector(getMostRecentOverviewPage);
+  const isBtcAccount = useMemo(
+    () => isBtcMainnetAddress(address) || isBtcTestnetAddress(address),
+    [address],
+  );
+
+  const rejectAction = useCallback(async (completed: boolean) => {
+    await onActionComplete({ success: completed });
+  }, []);
 
-  const onCreateAccount = useCallback(async (name: string) => {
-    dispatch(setAccountLabel(address, name));
-    await onActionComplete(true);
+  const onCreateAccount = useCallback(async (name?: string) => {
+    await onActionComplete({ success: true, name });
   }, []);
 
   const getNextAccountName = useCallback(
-    async (accounts: InternalAccount[]): Promise<string> => {
+    async (_accounts: InternalAccount[]): Promise<string> => {
       // if snapSuggestedAccountName exists, return it immediately
       if (snapSuggestedAccountName) {
         return snapSuggestedAccountName;
       }
 
-      // current Snap account has temporarily been created (this
-      // allow us to rename it afterward), so we should be able
-      // to get his current name:
-      const defaultAccountName: string = getAccountName(accounts, address);
-      // if defaultAccountName is truthy, return it immediately
-      if (defaultAccountName) {
-        return defaultAccountName;
-      }
+      const nextAccountName = await getNextAvailableAccountName(
```

Comment:
We no longer temporarily persist the account thats to be created. We can just get the next available name by using this action.

Reply:
I much prefer this pattern! Well done 🧠